ivy/functional/ivy/gradients.py

```diff
@@ -210,15 +209,24 @@ def execute_with_gradients(func, xs, retain_grads=False):
 
 
 @to_native_arrays_and_back
-def adam_step(dcdws, mw, vw, step, beta1=0.9, beta2=0.999, epsilon=1e-7):
+@handle_nestable
+def adam_step(
+    dcdw: Union[ivy.Array, ivy.NativeArray],
+    mw: Union[ivy.Array, ivy.NativeArray],
+    vw: Union[ivy.Array, ivy.NativeArray],
+    step: Union[int, float],
+    beta1=0.9,
+    beta2=0.999,
+    epsilon=1e-7,
+) -> ivy.Array:
     """Compute adam step delta, given the derivatives of some cost c with respect to ws,
     using ADAM update. `[reference]
 
     <https://en.wikipedia.org/wiki/Stochastic_gradient_descent#Adam>`_
 
     Parameters
     ----------
-    dcdws
+    dcdw(s)
         Derivates of the cost c with respect to the weights ws, [dc/dw for w in ws].
```

Comment:
this should be singular, for example "derivate of the cost c with respect to the weight w"